### PR TITLE
refactor: move operation "AddData" to "Draft.__setitem__"

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2fzxzjv SDK draft的setitem操作增加AddData operation](https://app.clickup.com/t/2fzxzjv)

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -194,6 +194,7 @@ def _from_pyarrow(cls: Type[_T], array: pa.StructArray, schema: pt.PortexType) -
         obj.schema = schema
         obj._columns = {}
         obj._column_names = []
+        obj.operations = None
```
Comment (linjiX) [inline graviti/dataframe/frame.py:197]: will it be better to set default value of `operations` to None, rather than set `operations` to None everywhere?

**Review by linjiX — APPROVED** [review]

File: graviti/manager/draft.py [hunk 2]
```diff
@@ -83,6 +83,13 @@ def __setitem__(self, key: str, value: DataFrame) -> None:
         if is_replace:
             self.operations.append(DeleteSheet(key))
 
+        if value.operations is None:
```
Comment (AChenQ) [inline graviti/manager/draft.py:86]: What if operations is `[]`

Comment (AChenQ) [reply graviti/manager/draft.py:86]: operations is initialized to `[]` in `DataFrame._from_paging`

**Review by AChenQ — APPROVED** [review]

File: graviti/manager/draft.py [hunk 3]
```diff
@@ -83,9 +83,17 @@ def __setitem__(self, key: str, value: DataFrame) -> None:
...
+        else:
+            raise NotImplementedError(
+                "Not support assigning one DataFrame to multiple sheets."
+                " Please use method 'copy' first."
+            )
         self.operations.append(CreateSheet(key, value.schema.copy()))
 
     def __delitem__(self, key: str) -> None:
+        self[key].operations = None
         super().__delitem__(key)
```
Comment (linjiX) [inline graviti/manager/draft.py:97]: `super().pop(key)`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/draft.py (hunk 3 above)
Comment (graczhual) [reply graviti/manager/draft.py:97]: but `pop` will call `__delitem__` recursively

Comment (linjiX) [reply graviti/manager/draft.py:97]: will `super().pop()` call `Draft.__delitem__`?

Comment (graczhual) [reply graviti/manager/draft.py:97]: Yes, I tried and failed.

File: graviti/manager/draft.py [hunk 4]
```diff
@@ -83,9 +83,17 @@ def __setitem__(self, key: str, value: DataFrame) -> None:
...
+            value.operations = [AddData(value.copy())]
+        else:
+            raise NotImplementedError(
+                "Not support assigning one DataFrame to multiple sheets."
+                " Please use method 'copy' first."
+            )
         self.operations.append(CreateSheet(key, value.schema.copy()))
 
     def __delitem__(self, key: str) -> None:
+        self[key].operations = None
```
Comment (linjiX) [inline graviti/manager/draft.py:96]: can we use `del` here?
```python
df = self[key]
del df.operations
```

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**